contracts/Optimism_SpokePool.sol

```diff
@@ -44,37 +39,68 @@ contract Optimism_SpokePool is CrossDomainEnabled, SpokePoolInterface, SpokePool
         public
         override
         onlyFromCrossDomainAccount(crossDomainAdmin)
+        nonReentrant
     {
         _setCrossDomainAdmin(newCrossDomainAdmin);
     }
 
+    function setHubPool(address newHubPool) public override onlyFromCrossDomainAccount(crossDomainAdmin) nonReentrant {
+        _setHubPool(newHubPool);
+    }
+
     function setEnableRoute(
         address originToken,
         uint256 destinationChainId,
         bool enable
-    ) public override onlyFromCrossDomainAccount(crossDomainAdmin) {
+    ) public override onlyFromCrossDomainAccount(crossDomainAdmin) nonReentrant {
         _setEnableRoute(originToken, destinationChainId, enable);
     }
 
-    function setDepositQuoteTimeBuffer(uint64 buffer) public override onlyFromCrossDomainAccount(crossDomainAdmin) {
+    function setDepositQuoteTimeBuffer(uint64 buffer)
+        public
+        override
+        onlyFromCrossDomainAccount(crossDomainAdmin)
+        nonReentrant
+    {
         _setDepositQuoteTimeBuffer(buffer);
     }
 
     function initializeRelayerRefund(bytes32 relayerRepaymentDistributionProof)
         public
         override
         onlyFromCrossDomainAccount(crossDomainAdmin)
+        nonReentrant
     {
         _initializeRelayerRefund(relayerRepaymentDistributionProof);
     }
 
-    /**************************************
-     *         INTERNAL FUNCTIONS         *
-     **************************************/
+    function distributeRelayerRefund(
+        uint256 relayerRefundId,
+        MerkleLib.DestinationDistribution memory distributionLeaf,
+        bytes32[] memory inclusionProof
+    ) public override nonReentrant {
+        _distributeRelayerRefund(relayerRefundId, distributionLeaf, inclusionProof);
+
+        // If `distributionLeaf.amountToReturn` is positive, then send L2 --> L1 message to bridge tokens back.
+        // Note: This might not be an issue at all, but should this call go before or after the above internal method
+        // that sends tokens from this contract to relayers on the same network?
+        if (distributionLeaf.amountToReturn > 0) {
+            // Do we need to perform any check about the last time that funds were bridged from L2 to L1?
+            IL2ERC20Bridge(Lib_PredeployAddresses.L2_STANDARD_BRIDGE).withdrawTo(
+                distributionLeaf.l2TokenAddress, // _l2Token. Address of the L2 token to bridge over.
+                hubPool, // _to. Withdraw, over the bridge, to the l1 pool contract.
+                distributionLeaf.amountToReturn, // _amount. Send the full balance of the deposit box to bridge.
+                6_000_000, // _l1Gas. Unused, but included for potential forward compatibility considerations
+                "" // _data. We don't need to send any data for the bridging action.
+            );
```

Comment:
nit: should we just put this bit of special logic in a virtual internal method that's called by the base class? That way the more generic logic could all live in the base class and this dispatch bit would be the only part that's custom per chain?

Reply:
Great idea, done: https://github.com/across-protocol/across-smart-contracts-v2/pull/24/commits/833d32056b8c31cf6cc33f144c4d8f77a4af903f